website/server/controllers/api-v3/groups.js

```diff
@@ -561,6 +561,12 @@ api.joinGroup = {
       }
     }
 
+    if (group.type === 'guild') {
```

Comment:
I don't actually know how to write tests. Or even where to start for that. If you can point me in the right direction, I'd be happy to give it a shot though.